workflows/src/main/kotlin/com/r3/corda/lib/ci/SignedKeyUtilities.kt

```diff
@@ -0,0 +1,114 @@
+package com.r3.corda.lib.ci
+
+import net.corda.core.CordaInternal
+import net.corda.core.crypto.DigitalSignature
+import net.corda.core.crypto.SignedData
+import net.corda.core.crypto.sha256
+import net.corda.core.internal.VisibleForTesting
+import net.corda.core.node.ServiceHub
+import net.corda.core.serialization.CordaSerializable
+import net.corda.core.serialization.SerializedBytes
+import net.corda.core.serialization.serialize
+import net.corda.core.utilities.OpaqueBytes
+import java.security.PublicKey
+import java.security.SignatureException
+import java.util.*
+
+/**
+ * Unique arbitrary identifier when a key-generating node is creating a [SignedOwnershipClaim].
+ */
+typealias OwnershipClaim = OpaqueBytes
+
+/**
+ * Generates and returns a signed [SignedOwnershipClaim] which contains data on the newly generated [PublicKey] to be associated
+ * with the provided external ID.
+ *
+ * @param serviceHub The [ServiceHub] of the node which requests a new public key
+ * @param uuid The external ID to be associated with the new [PublicKey]
+ */
+@CordaInternal
+@VisibleForTesting
+fun createSignedOwnershipClaimFromUUID(serviceHub: ServiceHub, nonce: OwnershipClaim, uuid: UUID): SerializedSignedOwnershipClaim<SignedOwnershipClaim> {
+    require(nonce.sha256().size == 32)
+    val nodeParty = serviceHub.myInfo.legalIdentities.first()
+    val newKey = serviceHub.keyManagementService.freshKey(uuid)
+    val signedOwnershipClaim = SignedOwnershipClaim(nonce, newKey)
```

Comment:
should the UUID be part of the ownership claim? (i.e. should it be signed over?)